metadata-service/services/src/main/java/com/linkedin/metadata/recommendation/candidatesource/EntitySearchAggregationSource.java

```diff
@@ -116,6 +117,11 @@ public List<RecommendationContent> getRecommendations(@Nonnull Urn userUrn,
         .collect(Collectors.toList());
   }
 
+  protected Filter buildAggregationFilter() {
```

Comment:
can we just have a method that liimits the entity indexes which are used for the query? 

then the base class can just return a set of entity type names. what do you think?

Reply:
Our Elastic SearchDAO is not prepared to received a set of entity names, it's either a specific one or null (which is all). It would be a greater lift.